java-client/src/main/java/org/opensearch/client/opensearch/_types/query_dsl/NeuralQuery.java

```diff
@@ -0,0 +1,203 @@
+import javax.annotation.Nullable;
+import org.opensearch.client.json.JsonpDeserializable;
+import org.opensearch.client.json.JsonpDeserializer;
+import org.opensearch.client.json.JsonpMapper;
+import org.opensearch.client.json.ObjectBuilderDeserializer;
+import org.opensearch.client.json.ObjectDeserializer;
+import org.opensearch.client.util.ApiTypeHelper;
+import org.opensearch.client.util.ObjectBuilder;
+
+@JsonpDeserializable
+public class NeuralQuery extends QueryBase implements QueryVariant {
+
+    private final String field;
+    private final String queryText;
+    private final int k;
+    @Nullable
+    private final String modelId;
+
+    private NeuralQuery(NeuralQuery.Builder builder) {
+        super(builder);
+
+        this.field = ApiTypeHelper.requireNonNull(builder.field, this, "field");
+        this.queryText = ApiTypeHelper.requireNonNull(builder.queryText, this, "queryText");
+        this.k = ApiTypeHelper.requireNonNull(builder.k, this, "k");
+        this.modelId = builder.modelId;
+    }
+
+    public static NeuralQuery of(Function<NeuralQuery.Builder, ObjectBuilder<NeuralQuery>> fn) {
+        return fn.apply(new NeuralQuery.Builder()).build();
+    }
+
+    /**
+     * Query variant kind.
+     *
+     * @return The query variant kind.
+     */
+    @Override
+    public Query.Kind _queryKind() {
+        return null;
+    }
+
+    /**
+     * Required - The target field.
+     *
+     * @return The target field.
+     */
+    public final String field() {
+        return this.field;
+    }
+
+    /**
+     * Required - Search query text.
+     *
+     * @return Search query text.
+     */
+    public final String queryText() {
+        return this.queryText;
+    }
+
+    /**
+     * Required - The number of neighbors to return.
+     *
+     * @return The number of neighbors to return.
+     */
+    public final int k() {
+        return this.k;
+    }
+
+    /**
+     * Builder for {@link NeuralQuery}.
+     */
+
+    /**
+     * Optional - The model_id field if the default model for the index or field is set.
+     * Required - The model_id field if there is no default model set for the index or field.
+     *
+     * @return The model_id field.
+     */
+    @Nullable
+    public final String modelId() {
+        return this.modelId;
+    }
+
+    @Override
+    protected void serializeInternal(JsonGenerator generator, JsonpMapper mapper) {
+        generator.writeStartObject(this.field);
+
+        super.serializeInternal(generator, mapper);
+
+        generator.write("query_text", this.queryText);
+
+        if (this.modelId != null) {
+            generator.write("model_id", this.modelId);
+        }
+
+        generator.write("k", this.k);
+
+        generator.writeEnd();
+    }
+
+    /**
+     * Builder for {@link NeuralQuery}.
+     */
+    public static class Builder extends QueryBase.AbstractBuilder<NeuralQuery.Builder> implements ObjectBuilder<NeuralQuery> {
+        @Nullable
```

Comment:
Only `modelId` is supposed to be nullable, others are required

Reply:
fixed